src/actix/api/snapshot_api.rs

```diff
@@ -70,8 +71,31 @@ pub async fn do_get_full_snapshot(
     snapshot_name: &str,
 ) -> Result<NamedFile, HttpError> {
     access.check_global_access(AccessRequirements::new())?;
-    let file_name = get_full_snapshot_path(toc, snapshot_name).await?;
-    Ok(NamedFile::open(file_name)?)
+    let snapshots_storage_manager = toc.get_snapshots_storage_manager().await;
+    match snapshots_storage_manager {
+        SnapshotStorageManager::LocalFS(_) => {
+            let snapshot_path = get_full_snapshot_path(toc, snapshot_name).await?;
+            Ok(NamedFile::open(snapshot_path)?)
+        }
+        SnapshotStorageManager::S3(_) => {
```

Comment:
As well, should be implementation-specific. Exposing internal logic of the snapshot manager into user code means we are breaking abstraction